azkaban-executor/azkaban-emr-jobtype/src/main/java/uk/gov/dwp/dataworks/azkaban/jobtype/EMRStep.java

```diff
@@ -143,9 +145,9 @@ public void run() throws Exception {
 
         AmazonElasticMapReduce emr = AmazonElasticMapReduceClientBuilder.standard().withRegion(awsRegion).build();
 
-        String clusterId = getClusterId(emr);
+        String clusterId = getClusterId(emr, AWSLambdaClientBuilder.defaultClient(), this.getSysProps(), this.getJobProps(), killed, getLog());
```

Comment:
I'm going to assume that passing in all these global datasets are to make integration tests easier, otherwise it just mess'es up the method signature!

Reply:
Yes, its purely to allow a unit test.